commands/poll.js

```diff
@@ -0,0 +1,333 @@
+const { SlashCommandBuilder } = require("@discordjs/builders");
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  ComponentType,
+  EmbedBuilder,
+} = require("discord.js");
+const dotenv = require("dotenv");
+
+dotenv.config();
+
+const MAX_OPTIONS = 10;
+const OPTION_EMOJIS = [
+  "1️⃣",
+  "2️⃣",
+  "3️⃣",
+  "4️⃣",
+  "5️⃣",
+  "6️⃣",
+  "7️⃣",
+  "8️⃣",
+  "9️⃣",
+  "🔟",
+];
+const CLIENT_ID = process.env.DISCORD_CLIENT_ID;
+const DELAY = 3;
+
+const getTopVotedOption = async (message, interaction) => {
+  const initialResults = await Promise.all(
+    message.reactions.cache.map(async (reaction) => {
+      const emojiName = reaction.emoji.name;
+      const reactionUserIds = await reaction.users.fetch();
+
+      // excludes bot
+      const reactionUsernames = await Promise.all(
+        [...reactionUserIds.keys()]
+          .filter((userId) => userId !== CLIENT_ID)
+          .map(async (userId) => {
+            const user = await interaction.client.users.fetch(userId);
+            if (!user.bot) {
+              return user.username;
+            }
+          })
+      );
+
+      const reactionCount = reactionUsernames.length;
+
+      return {
+        emoji: emojiName,
+        count: reactionCount,
+        reactors: reactionUsernames,
```

Comment:
Minor: I wouldn't really name the users who reacted with a specific reaction "reactors". It's more intuitive to just call it `users` or `usernames`. The latter might be a better option given that this would be a list of usernames.

```suggestion
        usernames: reactionUsernames,
```